### PR TITLE
Added API to access editor markers

### LINKED ISSUE #30: Access to Markers?
I'm killing myself trying to figure out how to get access to the info/warning/error markers created by the language linters. Is this possible?

I've found how I can access decorations (IModel.onDidChangeDecorations and a few get... methods in ITextModelWithDecorations). I can filter for decorations that are "IModelDecorationsChangedEventDecorationData.isForValidation), but that doesn't give me access to the marker.

I've found how I can SET markers (monaco.editor.setModelMarkers).

Am I missing something, or are the markers currently not exposed? Is there a preferred method of enumerating errors and warnings?

### PR DESCRIPTION
**Pull request by electroma**
As of now there is no public API to access markers in monaco-editor. This PR adds method `editor.getModelMarkers`. Should resolve https://github.com/Microsoft/monaco-editor/issues/30.

### CODE REVIEW

**Comment by jrieken** [comment]
I'd say that for reading `IMarkerData` isn't the right interface because it lacks owner and resource fields. That why there also is [`IMarker`](https://github.com/Microsoft/vscode/blob/master/src/vs/platform/markers/common/markers.ts#L49) with more information

File: src/vs/monaco.d.ts [hunk 1]
```diff
@@ -834,6 +834,17 @@ declare module monaco.editor {
...
 	/**
+	 * Get markers for owner ant/or resource
+	 * @returns {IMarkerData[]} list of markers
+	 * @param filter
+	 */
+	export function getModelMarkers(filter: {
+		owner?: string;
+		resource?: Uri;
+		take?: number;
+	}): IMarkerData[];
```
Comment (jrieken) [inline src/vs/monaco.d.ts]: Should return `IMarker` or enforce a resource in the get-call

**Review by jrieken — CHANGES_REQUESTED** [review]

File: src/vs/monaco.d.ts (hunk 1 above)
Comment (electroma) [inline src/vs/monaco.d.ts]: @jrieken thank you for spotting it! I've changed return to `IMarker`.

**Comment by electroma** [comment]
@jrieken i had to reorder declarations in https://github.com/Microsoft/vscode/blob/master/src/vs/platform/markers/common/markers.ts. Otherwise `monaco.d.ts` is not generated correctly.
The root cause is this line https://github.com/Microsoft/vscode/blob/master/build/monaco/api.ts#L142. 
I can raise separate issue to it.

**Comment by jrieken** [comment]
Thanks. looking good

**Comment by belohlavek** [comment]
Could it be that this change isn't included in the declaration file at https://github.com/Microsoft/vscode/tree/master/build/monaco?

**Comment by electroma** [comment]
@belohlavek i did update recipe and method `getModelMarkers` presents in generated https://github.com/Microsoft/vscode/blob/master/src/vs/monaco.d.ts#L841

**Pull request merged**